### ISSUE TITLE
Bug in S1 ARD slope correction

### ISSUE THREAD

**Issue by adugnag** [adugnag/gee_s1_ard#4]
There was an error in the calculation of the aspect and heading angle. The original implementation in Hoekman and Reiche 2015 expects angles from -180 to 180, while the aspect computed in GEE is from 0 to 360. In addition, the aspect angle on GEE seems downhill directed, while it should be uphill.

As a result, the correction factor needs to be multiplied by gamma0 instead of being divided. All this is corrected in the updated version.

**Comment by villemaire** [comment]
Hi,
Good and very useful work, thank you very much !
I would like to point to you a typo in today's version of "terrain_flattening.js"
Lines 77 and 80: 
The variable "elevation" is not previously declared. 
I guess that you meant to write "DEM" instead ?
Again, thanks for everything !
cheers

**Comment by adugnag** [comment]
Indeed, thanks for pointing it out!

**Comment by villemaire** [comment]
Hi Adugna,

In our test case, we encountered another issue with this week version of "terrain_flattening.js"

Your fix did a very good job, but some annoying artefacts appeared in our regions of interest.
These areas are characterized by lots of topographic elevation changes.
I found that bilinear or bicubic resampling of the dem seems to fix the problem. I also matched the scale of the dem to the same as the imagery (10m), but I'm not sure if it's required.

so at line 58, this is what works for us:
var elevation = DEM.resample('bilinear').reproject({crs:proj, scale:10}).clip(geom);
or
var elevation = DEM.resample('bicubic').reproject({crs:proj, scale:10}).clip(geom);

here's two examples to see for yourself :

 // add these two polygons to script "s1_ard.js"
var issue_artefacts_west_coast =  ee.Geometry.Polygon(
        [[[-125.95654653016003, 51.12119016229345],
          [-125.95654653016003, 51.10195321414363],
          [-125.90727971497448, 51.10195321414363],
          [-125.90727971497448, 51.12119016229345]]], null, false);

var issue_artefacts_grands_jardins = ee.Geometry.Polygon(
        [[[-70.49702756098367, 47.95723288536608],
          [-70.49702756098367, 47.929060236124705],
          [-70.44261090448953, 47.929060236124705],
          [-70.44261090448953, 47.95723288536608]]], null, false);

// and set variable "geometry" to one or the other of these two polygons:
geometry= issue_artefacts_grands_jardins;
//geometry= issue_artefacts_west_coast;

// On line 143 of "s1_ard.js", set zoom level to 14, you'll see the artefacts
Map.centerObject(parameter.GEOMETRY, 14);

I hope this helps, you might think of another way of fixing this, and, again, your functions are very helpful and the results are very good ! 
thanks for everything !
